src/coalescenceml/integrations/azure_datalake/artifact_store/azure_artifact_store.py

```diff
@@ -0,0 +1,83 @@
+from ast import Tuple
+import os
+from pathlib import Path
+from adlfs import AzureBlobFileSystem
+
+from coalescenceml.artifact_store.base_artifact_store import BaseArtifactStore, PathType
+from typing import (
+    Any,
+    Callable,
+    Iterable,
+    List,
+    Optional,
+)
+
+class AzureArtifactStore(BaseArtifactStore):
+    """ Artifact Store for Azure Datalake artifacts""" 
+
+    abfs: Optional[AzureBlobFileSystem] = None
+    account_name: str
+
+    def _ensure_filesystem_exists(self):
+        if not self.abfs:
+            self.abfs = AzureBlobFileSystem(account_name=self.account_name, anon=False)
+        return self.abfs
+
+    def open(self, name: PathType, mode: str = "rb") -> Any:
+        """ Open a file at""" 
+        self._ensure_filesystem_exists()
+        return self.abfs._open(name, mode=mode)
+    
+    def copyfile(self, src: PathType, dst: PathType, overwrite: bool = False) -> None:
```

Comment:
You need to ensure the file system exists. Another way to do this is assign a property “filesystem” which does the check or returns the file system.

Reply:
This would look like:

```python
@property
def filesystem(self):
    if not self.abfs:
        self.abfs = …
    return self.abfs
```

Then everywhere that uses `self.abfs` would instead use `self.filesystem`